### DISCUSSION TITLE
Bug? Namedtuple int incompatible with int property.

### DISCUSSION THREAD

**Discussion by bukzor** [microsoft/pyright#8679] [Q&A] [answered]
I can't see any incompatibility with these base classes, since they both define `attr` with no setter and no deleter.

Is this a bug or a feature?

```python
# pyright: strict
class Protocol:
    @property
    def attr(self) -> object:
        raise NotImplementedError
from typing import NamedTuple
class Mixin2(NamedTuple): attr: object
class Attempt2(Protocol, Mixin2): pass
# └╴E  Base classes for class "Attempt2" define variable "attr" in incompatible way Pyright (reportIncompatibleVariableOverride) [12, 7]
```

**Comment by erictraut** [comment]
This is by design. You have declared the type of `A.i`, but you haven't declared the type of `B.i`. It has an inferred type.

If you add a type declaration to `B.i`, as you have with `D.i`, you will see an error.

**Reply by bukzor** [reply]
Sorry I've updated the question because I saw that problem just now.

**Reply by erictraut** [reply, marked as answer]
A property is not the same as a normal attribute. In your example, the property doesn't support setting or deleting. Even if it did, the behavior would differ if you accessed the property from the class rather than a class instance. These declarations are not compatible, hence the error.

**Reply by bukzor** [reply]
This is a pretty idiomatic way to declare an abstract attribute. How would I accomplish this under strict pyright?
It's not as simple as `attr: object` because then pyright complains it's never initialized.

**Reply by erictraut** [reply]
If you want it to be type safe, you should avoid combining declarations of incompatible types. If you don't care about type safety in this particular case, you can suppress the error.

**Reply by bukzor** [reply]
Yes, I'd like to avoid combining declarations of incompatible types. I can edit the definition of `Protocol.attr`, but it's not obvious how to avoid this problem and avoid a "not initialized" error simultaneously.

```python
class Protocol: attr:object
# └╴E  Instance variable "attr" is not initialized in the class body or __init__ method Pyright (reportUninitializedInstanceVariable) [1, 17]
from typing import NamedTuple
class Mixin2(NamedTuple): attr: object
class Attempt2(Protocol, Mixin2): pass
```

**Reply by bukzor** [reply]
It seems like pyright is satisfied with this. I don't know all the repercussions of adding `typing.Protocol` as a base class though.

```python
# pyright: strict
from typing import Protocol

class MyProtocol(Protocol):
    attr: int

    def attr_x2(self) -> int:
        return self.attr * 2


from typing import NamedTuple
class Mixin2(NamedTuple): attr: int
class Attempt2(MyProtocol, Mixin2): pass
```
